Utilities/BoozerXForms/init_from_spec.py

```diff
@@ -0,0 +1,269 @@
+    xm        = d.output.im
+    xn        = d.output.in_
+    xm_nyq    = xm
+    xn_nyq    = xn
+    ns_in     = int(2*nvol-1) 
+    s_in      = np.ndarray(ns_in, dtype = np.float64)
+    s_in[0]   = d.output.tflux;
+    iota      = np.array([d.output.lambdamn[1][0][0]])
+    rmnc      = np.array([list(d.output.Rbc[1,:])]).transpose()
+    rmns      = np.array([])
+    zmnc      = np.array([])
+    zmns      = np.array([list(d.output.Zbs[1,:])]).transpose()
+    
+    # build lmns
+    lambdamn  = d.output.lambdamn[1][:].transpose()
+    xms       = d.output.ims
+    xns       = d.output.ins
+    mns       = d.output.mns
+    lmns      = np.zeros([mnmax,ns_in])
+    lmnc      = np.array([])
+    
+    for ii in range (0,mns):
+        mm = xms[ii]
+        nn = xns[ii]
+        
+        if mm==0 and nn==0: #mode (0,0) is zero
+            continue
+        if mm>mpol-1 or mm<0:
+            continue
+        if nn<-ntor*nfp or nn>ntor*nfp:
+            continue
+        
+        for jj in range(0,mnmax):
+            if mm==xm[jj] and nn==xn[jj]:
+                lmns[jj] = lambdamn[ii]
+                continue
+                           
+    
+    bsubumnc = np.array([list(d.output.Btemn[1,:])]).transpose()
+    bsubvmnc = np.array([list(d.output.Bzemn[1,:])]).transpose()
+    bsubumns = np.array([])
+    bsubvmns = np.array([])
+    mboz     = np.max(xm)
+    nboz     = int(1/nfp * np.max(xn))
+    aspect   = np.nan
+    toroidal_flux = d.input.physics.phiedge 
+    
+    # modB computation 
+    Nt   = d.grid.Nt
+    Nz   = d.grid.Nz
+    sarr = np.linspace(0,1,2)
+    tarr = np.linspace(0,2*np.pi,Nt)
+    zarr = np.linspace(0,2*np.pi/nfp,Nz)  
+    
+    Bcontrav       = d.get_B(lvol = 0, sarr = sarr, tarr = tarr, zarr = zarr)
+    [R, Z, jac, g] = d.get_grid_and_jacobian_and_metric(lvol = 0, sarr = sarr, tarr= tarr, zarr = zarr)
+    
+    modB     = d.get_modB(Bcontrav, g)
+    modBsurf = modB[1][:][:]
+    modBcos  = np.zeros(np.shape(modBsurf))
+    K        = 2*np.pi**2/nfp
+    Bmn_     = np.zeros((mpol,2*ntor+1))
+      
+    for m in range (0,mpol):
+        for n in range (-ntor,ntor+1):
+            if m == 0 and n<0:
+                continue
+            for line in range (0,Nt):
+                for column in range (0,Nz):
+                    modBcos[line,column] = modBsurf[line][column]* np.cos(np.double(m)*tarr[line] - np.double(n)*np.double(nfp)*zarr[column])
+
+            tmp = np.trapz(modBcos, x = tarr, axis = 0)
+            Bmn_[m,n+ntor] = 1/K*np.trapz(tmp, x=zarr)
+            
+    
+    Bmn_[0,ntor] = 1/2*Bmn_[0,ntor]
+    
+    
+    Bmnc       = np.zeros((mnmax,ns_in))
+    jj = 0 #index of surface
+    for ii in range(0,mnmax):
+        m = xm[ii]
+        n = int(xn[ii] / nfp)
+        Bmnc[ii,jj] = Bmn_[m,n+ntor]
+        
+    Bmns = np.array([])
+
+
+    ##########################################################
+    # initialisation de l'instance b de la classe booz_xform # 
```

Comment:
Comment should be in English, please :-)

Reply:
I'll do it ! My bad :)